### PR TITLE
[CI] Move all ROCm jobs to periodic frequency

### PR DESCRIPTION
**Pull request by jithunnair-amd**
`inductor` and `rocm` workflows are the major contributors to the CI load on ROCm CI at the moment, resulting in huge backlogs: https://github.com/pytorch/pytorch/pull/131489#issue-2425804464

* Move rocm.yml to cron frequency
* Move ROCm CI jobs from inductor.yml to inductor-rocm.yml
* Introduce `ciflow/inductor-rocm` as PR label to manually invoke inductor jobs for ROCm (no automatic invoking to limit CI load)
* After this PR, only `trunk` workflow jobs for ROCm will run on every commit and PR merge, but since they take 45min*3 time on average, I decided to leave them as-is since it will provide us some basic insulation against ROCm breakage.

cc @jeffdaily @sunway513 @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### CODE REVIEW

**Review by clee2000 — APPROVED** [review]
Looks good but please fix lint before merging

**Review by atalman — APPROVED** [review]

File: .github/workflows/inductor-rocm.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: inductor-rocm
+
+on:
+  schedule:
+    # We have several schedules so jobs can check github.event.schedule to activate only for a fraction of the runs.
+    # Also run less frequently on weekends.
+    - cron: 45 0,8,16 * * 1-5
```
Comment (huydhn) [inline .github/workflows/inductor-rocm.yml:7]: Do we need so many schedule here?  They come from the current periodic workflow, in this case I think we just want to run every 4 hours and keep the daily mem leak check and rerun disabled tests one

**Review by huydhn — APPROVED** [review]
LGTM! One small suggestion to simplify the list of schedules a bit

**Comment by huydhn** [comment]
cc @clee2000 @atalman I notice this huge stack https://github.com/pytorch/pytorch/pull/131564, we know that similar ones had caused long queue across the board in the past.

File: .github/workflows/inductor-rocm.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+name: inductor-rocm
+
+on:
+  schedule:
+    # We have several schedules so jobs can check github.event.schedule to activate only for a fraction of the runs.
+    # Also run less frequently on weekends.
+    - cron: 45 0,4,8,12,16,20 * * 1-5
+    - cron: 45 4,12 * * 0,6
```
Comment (jithunnair-amd) [inline .github/workflows/inductor-rocm.yml:8]: @huydhn I consolidated the schedules, but would prefer to keep the weekends lighter to not strain the CI unnecessarily.

**Pull request closed without merging**